### PR TITLE
Fixes #147: Carousel Image Alignment

### LINKED ISSUE #147: general Image/Slider Alignment relative to surrounding content
I don't know what the designs look like so feel free to close this issue. 

Slider images have a gutter and thus add extra initial/end slide padding.

| telegram-cloud-photo-size-1-5044150914482547440-y | telegram-cloud-photo-size-1-5044150914482547438-y | telegram-cloud-photo-size-1-5044150914482547437-y |
|------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|

Then there's this:

| telegram-cloud-photo-size-1-5044150914482547439-y |   |   |
|------------------------------------------------------------------------------------------------------------------------------------------------------------|---|---|

### PR DESCRIPTION
**Pull request by morganick**
# Description

[Trello Card]()

Fixes #147 

The carousels across the app were not precise with their scrolling. Sometimes we'd over animate, over scroll, etc. This PR solves those issues. We now have a single place where we can adjust the size of the carousel cards with respect to the screen size so we can stop _mathing_ so hard with each change.

This should reduce churn when working with this component.

I also made some adjustments to the `BoxShadow` component to make it "just work" where ever you used it.

At some point, I think these are good component candidates for extraction. Then we can make them more configurable and easy to work with.

# Graphics

| iOS            | Android            |
| -------------- | ------------------ |
| ios_screenshot | android_screenshot |

# Checklist:

- [ ] I have done a thorough self-review of my code
- [ ] I have tested with a screen reader and font-scaling turned on and added necessary accessibility features
- [ ] I have run tests and linter

### CODE REVIEW

**Review by mazenchami — COMMENTED** [review]
Tested on both iOS/Android devices i have and it looks great! Great job! Just a couple comments

File: app/screens/TalkDetailsScreen/WorkshopDetailsScreen.tsx [hunk 1]
```diff
@@ -130,5 +135,8 @@ const $subtitle: TextStyle = {
 }
 
 const $headingContainer: ViewStyle = {
+  ...$contentSpacing,
   marginBottom: spacing.extraLarge,
 }
+
+const $carouselContainer: ViewStyle = {}
```
[the remark this answers is not in the document]
Comment (morganick) [reply app/screens/TalkDetailsScreen/WorkshopDetailsScreen.tsx]: 👍 addressed in [b9f2495](https://github.com/infinitered/ChainReactApp2023/pull/172/commits/b9f2495ceafaa61b923f73a184922f13b29efedd)

File: app/components/carousel/Carousel.tsx [hunk 2]
```diff
@@ -24,8 +24,10 @@ export function Carousel(props: CarouselProps) {
   const scrollX = useSharedValue(0)
   const onScroll = useAnimatedScrollHandler((event) => (scrollX.value = event.contentOffset.x))
 
+  const endingSpacer = SCREEN_CONTENT_WIDTH - CAROUSEL_INTERVAL + SCREEN_CONTENT_GUTTER
```
[the remark this answers is not in the document]
Comment (morganick) [reply app/components/carousel/Carousel.tsx]: Totally agree, addressed in [b9f2495](https://github.com/infinitered/ChainReactApp2023/pull/172/commits/b9f2495ceafaa61b923f73a184922f13b29efedd)

**Review by mazenchami — APPROVED** [review]
Great work on this! This looks awesome! 🎉 🚀

**Pull request merged**